### PR TITLE
Added constructors and destructors for RgbdPlane

### PR DESCRIPTION
**Pull request by czgdp1807**
resolves https://github.com/opencv/opencv_contrib/issues/2277

### This pullrequest changes

This PR adds constructors and destructors for RgbdPlane class.

### CODE REVIEW

**Comment by czgdp1807** [comment]
I will add the definitions for each one of them once their API is approved. 
ping @Takugo @alalek  Please take a look and let me know if you want some modifications.

File: modules/rgbd/include/opencv2/rgbd.hpp [hunk 1]
```diff
@@ -379,6 +379,26 @@ namespace rgbd
...
+     * @param sensor_error_a_ coefficient of the sensor error with respect to the. 0 by default, 0.0075 for a Kinect
+     * @param sensor_error_b_ coefficient of the sensor error with respect to the. 0 by default
+     * @param sensor_error_c_ coefficient of the sensor error with respect to the. 0 by default
+     * @param method The method to use to compute the planes, by default RGBD_PLANE_METHOD_DEFAULT
+     */
+    RgbdPlane(int block_size, int min_size, double threshold, double sensor_error_a_ = 0,
+              double sensor_error_b_ = 0, double sensor_error_c_ = 0, int method =
+              RgbdPlane::RGBD_PLANE_METHOD_DEFAULT);
+
+    ~RgbdPlane();
```
Comment (alalek) [inline modules/rgbd/include/opencv2/rgbd.hpp]: Not defined anywhere.

Comment (czgdp1807) [reply modules/rgbd/include/opencv2/rgbd.hpp]: I will add all the necessary definitions once the above declarations are approved.

**Comment by czgdp1807** [comment]
Thanks for the comments and reviews. Is it possible to obtain the paper which was referred for implementing `RgbdPlane` class? If not available, please share any valid literature which can cited in the documentation.

**Comment by czgdp1807** [comment]
I have asked the question about references [here](https://answers.opencv.org/question/221491/references-for-rgbdplane/) and https://github.com/opencv/opencv_contrib/pull/10#issuecomment-552415536

File: modules/rgbd/include/opencv2/rgbd.hpp [hunk 2]
```diff
@@ -379,6 +379,26 @@ namespace rgbd
...
+     * @param min_size The minimum size of a cluster to be considered a plane
+     * @param threshold The maximum distance of a point from a plane to belong to it (in meters)
+     * @param sensor_error_a_ coefficient of the sensor error with respect to the. 0 by default, 0.0075 for a Kinect
+     * @param sensor_error_b_ coefficient of the sensor error with respect to the. 0 by default
+     * @param sensor_error_c_ coefficient of the sensor error with respect to the. 0 by default
+     * @param method The method to use to compute the planes, by default RGBD_PLANE_METHOD_DEFAULT
+     */
+    RgbdPlane(int block_size, int min_size, double threshold, double sensor_error_a_ = 0,
+              double sensor_error_b_ = 0, double sensor_error_c_ = 0, int method =
+              RgbdPlane::RGBD_PLANE_METHOD_DEFAULT);
```
Comment (alalek) [inline modules/rgbd/include/opencv2/rgbd.hpp]: > int method = RgbdPlane::RGBD_PLANE_METHOD_DEFAULT

Please move to the beginning of parameters list (methods define meaning of its parameters)

**Comment by Takugo** [comment]
@czgdp1807 I found some references in ..\rgbd\src\plane.cpp,
RgbdPlane() may be implemented referring to papers below:

> /** This is an implementation of a fast plane detection loosely inspired by
>  * Fast Plane Detection and Polygonalization in noisy 3D Range Images
>  * Jann Poppinga, Narunas Vaskevicius, Andreas Birk, and Kaustubh Pathak
>  * and the follow-up
>  * Fast Plane Detection for SLAM from Noisy Range Images in
>  * Both Structured and Unstructured Environments
>  * Junhao Xiao, Jianhua Zhang and Jianwei Zhang
>  * Houxiang Zhang and Hans Petter Hildre
>  */


It may help you.

> Thanks for the comments and reviews. Is it possible to obtain the paper which was referred for implementing `RgbdPlane` class? If not available, please share any valid literature which can cited in the documentation.

**Comment by czgdp1807** [comment]
I just defined the constructors/destructors and below is the result,
```python
Python 3.6.8 (default, Oct  7 2019, 12:59:55) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cv2
>>> rgbd_plane = cv2.rgbd.RgbdPlane_create(cv2.rgbd.RgbdPlane_RGBD_PLANE_METHOD_DEFAULT, 40, 1600, 0.01, 0, 0, 0)
>>> rgbd_plane
<rgbd_RgbdPlane 0x7f5b931f5c50>
>>> 
```
I will add the test for python bindings by tomorrow and will push the final changes here.

**Comment by czgdp1807** [comment]
ping @alalek Please take a look and let me know of any changes required. Thanks.

File: modules/rgbd/misc/python/test/test_rgbd.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+from tests_common import NewOpenCVTests
+
+OPENCV_TEST_DATA_PATH = os.environ('OPENCV_TEST_DATA_PATH')
+
+class rgbd_test(NewOpenCVTests):
+
+   def test_computeRgbdPlane(self):
+
+      depth_image = cv.imread(OPENCV_TEST_DATA_PATH + '/cv/rgbd/depth.png', cv2.IMREAD_ANYDEPTH)
```
Comment (alalek) [inline modules/rgbd/misc/python/test/test_rgbd.py]: Please use [these methods](https://github.com/opencv/opencv/blob/4.1.2/modules/python/test/tests_common.py#L29-L49) instead.

File: modules/rgbd/misc/python/test/test_rgbd.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                     [[-0.04455929, -0.87659335, -0.47916424,  3.93200684]],
+                     [[-0.21514639,  0.18835169, -0.95824611,  7.59479475]],
+                     [[-0.01006953, -0.86679155, -0.49856904,  4.01355648]],
+                     [[-0.00876531, -0.87571168, -0.48275498,  3.96768975]],
+                     [[-0.06395926, -0.86951321, -0.48975089,  4.08618736]],
+                     [[-0.01403128, -0.87593341, -0.48222789,  7.74559402]],
+                     [[-0.01143177, -0.87495202, -0.4840748 ,  7.75355816]]],
+                     dtype=numpy.float32)
+
+      self.assertEquals(planes_coeff, planes_coeff_expected)
```
Comment (alalek) [inline modules/rgbd/misc/python/test/test_rgbd.py]: This check doesn't look reliable between different platforms or optimization options.

Please [use](https://github.com/opencv/opencv/blob/4.1.2/modules/python/test/test_fitline.py#L66) norm(L2) with non-zero threshold.

**Comment by czgdp1807** [comment]
Thanks for pointing out. I will make changes to the tests.

File: modules/rgbd/misc/python/test/test_rgbd.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+from tests_common import NewOpenCVTests
+
+class rgbd_test(NewOpenCVTests):
+
+   def test_computeRgbdPlane(self):
+
+      depth_image = self.get_sample('/cv/rgbd/depth.png', cv.IMREAD_ANYDEPTH)
+      if depth_image is None:
+         raise unittest.SkipTest("Missing files with test data")
```
Comment (alalek) [inline modules/rgbd/misc/python/test/test_rgbd.py]: please use indentation with 4 spaces

Comment (czgdp1807) [reply modules/rgbd/misc/python/test/test_rgbd.py]: I think my VSCode settings were incorrect and indentation size was reset to 3. Corrected now.

File: modules/rgbd/src/odometry.cpp [hunk 6]
```diff
@@ -1009,6 +1009,28 @@ Ptr<DepthCleaner> DepthCleaner::create(int depth_in, int window_size_in, int met
   return makePtr<DepthCleaner>(depth_in, window_size_in, method_in);
 }
 
+RgbdPlane::RgbdPlane(int method, int block_size,
```
Comment (alalek) [inline modules/rgbd/src/odometry.cpp]: These definitions should be placed in plane.cpp

Comment (czgdp1807) [reply modules/rgbd/src/odometry.cpp]: Will do that in a new PR.

**Pull request merged**